src/actions/CryptoExchangeActions.js

```diff
@@ -376,15 +376,25 @@ export const getShapeShiftTokens = () => async (dispatch: Dispatch, getState: Ge
   } catch (error) {
     dispatch({ type: 'ON_AVAILABLE_SHAPE_SHIFT_TOKENS', data: { response: {}, totalSwaps } })
   }
+  setTimeout(() => {
+    const currentScene = Actions.currentScene
+    if (currentScene !== Constants.EXCHANGE_QUOTE_SCENE) {
+      return
+    }
+    dispatch(getShapeShiftTokens())
```

Comment:
good catch.

Reply:
fixed

Reply:
fixed

Reply:
Because you are using setTimeout, this could fire after the user is logged out. Please check to make sure all proper redux state is present before touching it.